src/inmanta/data/__init__.py

```diff
@@ -2298,13 +2636,158 @@ async def get_resources_for_version_raw(
         return resources
 
     @classmethod
-    async def get_latest_version(cls, environment: uuid.UUID, resource_id: ResourceIdStr) -> Optional["Resource"]:
+    async def get_latest_version(cls, environment: uuid.UUID, resource_id: m.ResourceIdStr) -> Optional["Resource"]:
         resources = await cls.get_list(
             order_by_column="model", order="DESC", limit=1, environment=environment, resource_id=resource_id
         )
         if len(resources) > 0:
             return resources[0]
 
+    @classmethod
+    def _get_released_resources_partial_base_query(cls):
+        """A partial query describing the conditions for selecting the latest released resources,
+        according to the model version number."""
+        return f"""
+            FROM (
+                SELECT DISTINCT ON (r.resource_id) r.resource_id, r.attributes, r.resource_version_id, r.resource_type,
+                    r.agent, r.resource_id_value, r.environment,
+                    (CASE WHEN
+                            (SELECT r.model < MAX(public.configurationmodel.version)
+                            FROM public.configurationmodel
+                            WHERE public.configurationmodel.released=TRUE
+                            GROUP BY environment)
```

Comment:
* Shouldn't this sub-query filter on environment ID as well? We should add a test case to assert that this query respects the environment isolation.
* Why is the `GROUP BY` required here?
* I think we should wrap the select query, which is part of the `CASE WHEN` condition, into an `EXISTS()`. This might increase the performance of the query, since it will make the query execute until the first row is returned and not until the end. But PostgreSQL might be smart enough to do this automatically. I don't know.

Reply:
I think this has to become:

```

```

Reply:
I think this has to become something like this:

```suggestion
                            r.model < (SELECT MAX(cm_max.version)
                                             FROM public.configurationmodel as cm_max
                                             WHERE cm_max.released=TRUE AND cm_max.environment = r.environment
                                             )
```

Reply:
The `GROUP BY` was left over from an earlier iteration, it's not necessary now, nice catch.
Added the appropriate filter and extended the test cases